packages/debug/src/browser/view/debug-configuration-select.tsx

```diff
@@ -0,0 +1,223 @@
+        return DebugConfigurationSelect.PICK + providerType;
+    }
+
+    protected parsePickValue(value: string): string {
+        return value.slice(DebugConfigurationSelect.PICK.length);
+    }
+
+    protected async resolveDynamicConfigurationPicks(providerType: string): Promise<DynamicPickItem[]> {
+        const configurationsOfProviderType =
+            (await this.manager.provideDynamicDebugConfigurations()).find(entry => entry.type === providerType);
+
+        const picks: DynamicPickItem[] = [];
+        if (!configurationsOfProviderType) {
+            return picks;
+        }
+
+        const { configurations } = configurationsOfProviderType;
+        for (const configuration of configurations) {
+            picks.push({
+                label: configuration.name,
+                configurationType: configuration.type,
+                request: configuration.request,
+                providerType
+            });
+        }
+        return picks;
+    }
+
+    protected async selectDynamicConfigFromQuickPick(providerType: string): Promise<void> {
+        const picks: DynamicPickItem[] = await this.resolveDynamicConfigurationPicks(providerType);
+
+        if (picks.length === 0) {
+            return;
+        }
+
+        const selected: DynamicPickItem | undefined = await this.quickInputService.showQuickPick(
+            picks,
+            {
+                placeholder: nls.localizeByDefault('Select Launch Configuration')
+            }
+        );
+
+        if (!selected) {
+            return;
+        }
+
+        const selectedConfiguration = {
+            name: selected.label,
+            type: selected.configurationType,
+            request: selected.request
+        };
+        this.manager.current = this.manager.find(selectedConfiguration, undefined, selected.providerType);
+    }
+
+    protected refreshDebugConfigurations = async () => {
+        const configsPerType = await this.manager.provideDynamicDebugConfigurations();
+        const providerTypes = [];
+        for (const { type, configurations } of configsPerType) {
+            if (configurations.length > 0) {
+                providerTypes.push(type);
+            }
+        }
+        this.setState({ providerTypes });
+    };
+
+    protected renderOptions(): React.ReactNode {
+        const options: React.ReactNode[] = [];
+
+        // Add non dynamic debug configurations
+        for (const config of this.manager.all) {
+            const value = InternalDebugSessionOptions.toValue(config);
+            options.push(<option key={value} value={value}>
+                {this.toName(config, this.props.isMultiRoot)}
+            </option>);
+        }
+
+        // Add recently used dynamic debug configurations
+        const { recentDynamicOptions } = this.manager;
+        if (recentDynamicOptions.length > 0) {
+            options.push(<option key={'recent-configs-sep'} disabled>{DebugConfigurationSelect.SEPARATOR}</option>);
```

Comment:
I believe this should be conditional on the presence of non-dynamic options. Otherwise it's possible to end up with an initial separator:

![image](https://user-images.githubusercontent.com/62660806/152612064-13380440-34a1-4a1f-8470-259542082202.png)

Unrelated to this PR, the styling of our `select` elements is pretty bad on MacOS (screenshot from Chrome).  I'll make an issue for that.